scripts/install_shiny_deps.R

```diff
@@ -0,0 +1,16 @@
+
+# Install dependencies declared by a Shiny app
+# Relies on the undocumented behavior that devtools::install_deps will work
+# on any directory containing a DESCRIPTION file, even if not an R package.
+
+path <- commandArgs(trailingOnly = TRUE)[[1]]
+
+devtools::install_deps(path)
+dep <- desc::desc_get_deps(file.path(path, "DESCRIPTION"))
+purrr::walk(
+	dep$package,
+	~if (system.file(package = .) == "") {
```

Comment:
An interesting way to check that a package is installed. Elsewhere in PEcAn (or at least in all the Dockerfiles), we tend to use `pkg %in% rownames(installed.packages())`. Any particular reason you're doing it this way here? Again, no need to change, just curious. Though it might be nice to add a quick comment that this is checking if a package is installed.

Also, does `devtools::install_deps` throw warnings about failure to install anything? If so, would it be easier to run that with warnings temporarily converted to errors?

Reply:
> Also, does devtools::install_deps throw warnings about failure to install anything?
No, alas, that's exactly what I'm trying to work around here. `install_deps` returns nothing and never throws warnings, just lists on stdout the packages it doesn't know how to install. My first draft used capture.output and grepped for the string "Not on CRAN", but that seemed awfully brittle.

> we tend to use pkg %in% rownames(installed.packages())
The docs for `installed.packages` specifically recommend against this usage -- and recommends `system.file` instead! -- because `installed.packages` is expensive and has to read *multiple* files from every package in the library. It usually hides the slowness by caching its result for the whole session, but since we use it here right after installing packages the cache is usually invalid and won't help.

> might be nice to add a quick comment that this is checking if a package is installed
This is a good call and I'll do it.

Reply:
explanatory comments added in #2130

Reply:
Huh, good to know! We should use this in the Dockerfiles and elsewhere at some point.